publish/run-all-rpms.sh

```diff
@@ -1,7 +1,47 @@
 #!/bin/bash
+. /etc/profile.d/enable-alice.sh
 
-# Check if variables are provided.
-[ X${DEBUG:-1} = X1 ] && set -x
+# enable-alice.sh doesn't work with set -e, so only enable it now.
+set -exo pipefail
 
-python -m cProfile -s cumtime ./aliPublish --config "aliPublish-rpms.conf" --debug --cache-deps-dir /tmp/pubdepscache sync-rpms >&2
-timeout 300 rclone sync --config /secrets/alibuild_rclone_config --transfers=10 --verbose local:/repo/RPMS/el7.x86_64/ rpms3:alibuild-repo/RPMS/el7.x86_64/ || true
+# The architecture should correspond to the architecture built using the
+# requested config files -- only these architectures' repos will be synced.
+arch=$1
+shift   # $@ now contains only the config files to read.
+
+# aliPublish needs these S3 secrets.
+# rclone gets them from /secrets/alibuild_rclone_config.
+. /secrets/aws_bot_secrets
+export AWS_ACCESS_KEY_ID AWS_SECRET_ACCESS_KEY
+
+# This script is in the same directory as aliPublish{,S3} in the ali-bot repo.
+cd "$(dirname "$0")"
+
+# Look for any canary files from Jenkins builds that finished since we last ran.
+# We'll remove these once we're done, so that the respective Jenkins jobs can
+# look for the RPMs they expect.
+# We build this file list now as aliPublish can take some time, and other builds
+# (that we don't want to notify yet) might finish in the meantime. We run
+# aliPublish only for one specific architecture, so we have
+# architecture-specific canary files under rpmstatus/$arch/.
+s3cmd ls "s3://alibuild-repo/rpmstatus/$arch/" | cut -b 32- > canaries.txt
+
+case "$arch" in
+  el8.*)
+    pip install boto3    # aliPublishS3 needs boto3.
```

Comment:
Can you put the dependency in `setup.py`?

Reply:
It's already there. However, `setup.py` (and `requirements.txt`) also has stuff we don't need here, like `python-ldap`, `PyGithub` and `klein`. If you prefer, I can put a `pip install -r ../requirements.txt` in the script instead, though.

Reply:
(We don't install ali-bot here, we just clone it, as we need the config files from it, and if we're cloning it already, we might as well just use the scripts from the repo.)